src/loaders/compute_tools/microtrait/microtrait.py

```diff
@@ -79,6 +81,25 @@ def _run_microtrait(genome_id: str, fna_file: Path, genome_dir: Path, debug: boo
         # Resource Acquisition:Substrate uptake:biopolymer transport,3,Biopolymer transport,Resource Acquisition:Substrate uptake:biopolymer transport,Resource Acquisition,count,3,production,2,3
         trait_counts_df = _r_table_to_df(trait_counts)
 
+        trait_unwrapped_rules_file = os.environ.get('MT_TRAIT_UNWRAPPED_FILE')
+        if trait_unwrapped_rules_file:
+            trait_unwrapped_rules_df = pd.read_csv(trait_unwrapped_rules_file, sep='\t')
+            trait_counts_df = trait_counts_df.merge(trait_unwrapped_rules_df,
+                                                    left_on=loader_common_names.MICROTRAIT_TRAIT_NAME,
+                                                    right_on=loader_common_names.SYS_TRAIT_ID,
+                                                    how='left')
+            trait_counts_df.drop(columns=[loader_common_names.SYS_TRAIT_ID], inplace=True)
+
+            genes_detected_table = _get_r_list_element(microtrait_result, 'genes_detected_table')
+            genes_detected_df = _r_table_to_df(genes_detected_table)
+            detected_genes_score = dict(zip(genes_detected_df[_GENE_NAME_COL], genes_detected_df[_GENE_SCORE_COL]))
+
+            trait_counts_df[loader_common_names.DETECTED_GENE_SCORE_COL] = trait_counts_df[
+                loader_common_names.UNWRAPPED_GENE_COL].apply(
+                lambda x: [{gene: gene_score} for gene, gene_score in detected_genes_score.items() if gene in str(x)])
```

Comment:
I would split the string and iterate over that instead of iterating over the entire detected genes table every time

Reply:
👍

Reply:
:+1: